src/components/data/kanban/kanban.tsx

```diff
@@ -220,359 +124,207 @@ export const Kanban: React.FC<KanbanProps> = ({
     </div>
   );
 };
 
-export type KanbanSectionProps = Omit<
-  React.ComponentProps<"section">,
-  "onClick"
-> & {
-  buttonLinkProps?: ButtonLinkProps;
-  buttonProps?: ButtonProps;
+export type KanbanSectionProps = {
   columnIndex: number;
+  items: React.ReactNode[];
   draggable?: boolean;
   dragIndex: [number, number] | null;
-  fieldset: FieldSet;
-  fieldsets: FieldSet[];
-  fieldsetIndex: number;
   labelSelectColumn?: string;
   labelMoveObject?: string;
-  objectList: AttributeData[];
-  renderPreview?: (attributeData: AttributeData) => React.ReactNode;
-  urlFields: (keyof KanbanSectionProps["objectList"][number])[];
-  onClick?: (event: React.MouseEvent, attributeData: AttributeData) => void;
   onDragOver: React.DragEventHandler;
   onDrop: React.DragEventHandler;
-  onObjectChange?: (
-    object: AttributeData,
+  moveObject: (
+    object: React.ReactNode,
     sourceColumnIndex: number,
     targetColumnIndex: number,
     sourceObjectIndex: number,
     targetObjectIndex: number,
   ) => void;
+  componentListState: KanbanColumn[];
 };
 
 export const KanbanSection: React.FC<KanbanSectionProps> = ({
-  buttonProps,
-  buttonLinkProps,
   columnIndex,
+  items,
   draggable,
   dragIndex,
-  fieldset,
-  fieldsets,
-  fieldsetIndex,
-  labelSelectColumn,
-  labelMoveObject,
-  objectList,
-  renderPreview,
-  urlFields,
-  onClick,
   onDragOver,
   onDrop,
-  onObjectChange,
+  moveObject,
+  labelSelectColumn,
+  labelMoveObject,
+  componentListState,
 }) => {
   const isDragging = Boolean(dragIndex && dragIndex[0] === columnIndex);
+
   return (
     <Column
-      className={isDragging ? "mykn-kanban__drop-target" : undefined}
+      span={1}
       direction="column"
       gap={true}
-      span={1}
+      className={isDragging ? "mykn-kanban__drop-target" : undefined}
       onDragOver={onDragOver}
       onDrop={onDrop}
       data-column-index={columnIndex}
     >
-      {fieldset[0] && (
-        <Body>
-          <H3>{fieldset[0]}</H3>
-        </Body>
-      )}
-      {objectList.map((o, index) => (
-        <React.Fragment key={index}>
-          {isDragging && dragIndex?.[1] === index && (
-            <div className="mykn-kanban__drop-indicator" aria-hidden />
-          )}
-          <KanbanItem
-            key={index}
-            buttonLinkProps={buttonLinkProps}
-            buttonProps={buttonProps}
-            draggable={draggable}
-            fieldset={fieldset}
-            fieldsets={fieldsets}
-            fieldsetIndex={fieldsetIndex}
-            labelSelectColumn={labelSelectColumn}
-            labelMoveObject={labelMoveObject}
-            object={o}
-            objectIndex={index}
-            objectList={objectList}
-            renderPreview={renderPreview}
-            urlFields={urlFields}
-            onClick={onClick}
-            onObjectChange={onObjectChange}
-          />
-        </React.Fragment>
+      {items.map((item, index) => (
+        <KanbanItem
+          key={index}
```

Comment:
Wouldn't this cause the keys per item to change when an item is dragged and dropped? 
https://react.dev/learn/rendering-lists#rules-of-keys

Maybe the ID of the item should be used as the key?

Reply:
These should actually remain as the index keys as we calculate which item it is by using the indexes of the array itself

Reply:
I am not sure I understand what you mean? :thinking:

Reply:
Discussed: an ID property was added to the items